### PR TITLE
Refactor QuickInfoProvider

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
It also fix the bug I introduced in one of my recent PRs: links are not shown for types that defined in signature files.

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs [hunk 1]
```diff
@@ -331,31 +352,33 @@ type internal FSharpQuickInfoProvider
...
-                          | false, true -> yield! targetDocumentation
-                          | false, false when implText.Equals (sigText, StringComparison.OrdinalIgnoreCase) -> yield! sigDocumentation
-                          | false, false -> 
+                        [ match getText targetDocumentation, getText sigDocumentation with
+                          | None, None -> ()
+                          | None, Some _ -> yield! sigDocumentation
+                          | Some _, None -> yield! targetDocumentation
+                          | Some implText, Some sigText when implText.Equals (sigText, StringComparison.OrdinalIgnoreCase) -> 
+                              yield! sigDocumentation
+                          | Some _, Some _ ->
```
Comment (cloudRoutine) [inline vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs]: I think this is cleaner
```fsharp
                        match implDocsEmpty, sigDocsEmpty with
                        | true, true -> Seq.empty
                        | true, false -> upcast sigDocumentation
                        | false, true -> upcast targetDocumentation
                        | false, false when implText.Equals (sigText, StringComparison.OrdinalIgnoreCase) -> upcast sigDocumentation
                        | false, false -> 
                          seq{    
                            yield! sigDocumentation
                            yield lineBreak
                            yield seperator
                            yield lineBreak
                            yield! targetDocumentation }
```
and converting the resizeArray to a list to cast it to a seq is unnecessary, just cast it straight to a seq

Comment (vasily-kirichenko) [inline vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs]: I don't agree, sorry.

**Comment by KevinRansom** [comment]
Sorry @vasily-kirichenko you got bit by @cloudRoutine's refactor.

**Comment by vasily-kirichenko** [comment]
Done.

**Pull request merged**